week-6/Homework/extra/2-slideshow/slideshow.js

```diff
@@ -0,0 +1,95 @@
+// Write your code here
+let techs = [
+  "pics/html.jpg",
+  "pics/css.jpg",
+  "pics/javascript.jpg",
+  "pics/node.jpg",
+  "pics/react.jpg",
+];
+let index = 0;
+
+function slideShow() {
+  renderIndex(index);
+}
+
+function forwardMove() {
+  index++;
+  if (index == techs.length) {
+    index = index - techs.length;
+  }
+  renderIndex(index);
+  renderImage(index);
+}
+
+function backMove() {
+  index--;
+  if (index < 0) {
+    index = index + techs.length;
+  }
+  renderIndex(index);
+  renderImage(index);
+}
+
+function renderIndex(i) {
+  document.querySelector("#indexPlace").textContent = i;
+}
+function renderImage(i) {
+  document.querySelector("#imgPlace").src = techs[i];
+}
+
+function autoBackMove() {
+  addRemainTime();
+  setTimer(backMoveWithTime);
+}
+
+function backMoveWithTime() {
+  backMove();
+  addRemainTime();
+}
+
+function forwardMoveWithTime() {
+  forwardMove();
+  addRemainTime();
+}
+
+function autoForwardMove() {
+  setTimer(forwardMoveWithTime);
+  addRemainTime();
+}
+
+function stopAuto() {
+  clearInterval(timerImage);
+  clearInterval(timerRemainTime);
+}
+
+let timerImage, timerRemainTime;
+function setTimer(func) {
+  timerImage = setInterval(func, 5000);
+}
+
+function addRemainTime() {
+  let timer = 5;
```

Comment:
This is where you might need to refactor your code a bit if you wanted the user to be able to change the time, as suggested in the extension task.

Reply:
Thanks for your feedback. I have added the ability to change the time for user. But I am not sure that it is the best solution because after changing interval user needs to run automatic movement again. Unfortunately, I could not find way to know which function was running at the changing moment in order to make it working again. Could you advise something about it please?  Thanks for your assistance.